### PR TITLE
[StructuralMechanicsApplication]Adding process to compute solid-shell thickness

### PR DESCRIPTION
**Pull request by loumalouomega**
Process added + test

Improved script for SPRISM elements

### CODE REVIEW

**Comment by loumalouomega** [comment]
I corrected some bugs on the ShellToSolidShellProcess

File: applications/StructuralMechanicsApplication/custom_processes/solid_shell_thickness_compute_process.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        // We get the condition geometry
+        GeometryType& r_this_geometry = it_elem->GetGeometry();
+
+        if (r_this_geometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Prism3D6) {
+            connectivity_map.insert({r_this_geometry[0].Id(), r_this_geometry[3].Id()});
+            connectivity_map.insert({r_this_geometry[1].Id(), r_this_geometry[4].Id()});
+            connectivity_map.insert({r_this_geometry[2].Id(), r_this_geometry[5].Id()});
+        } else if (r_this_geometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Hexahedra3D8) {
+            // NOTE: It will be assumed that it keeps the default connectivity of GiD, please check it or it will return whatever
+//             KRATOS_WARNING_ONCE("SolidShellThickComputeProcess") << "It will be assumed that it keeps the default connectivity of GiD, please check it or it will return whatever" << std::endl;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_processes/solid_shell_thickness_compute_process.cpp:52]: Needed?

File: applications/StructuralMechanicsApplication/python_scripts/sprism_process.py [hunk 2]
```diff
@@ -2,41 +2,140 @@
...
     return SPRISMProcess(Model, settings["Parameters"])
 
-class SPRISMProcess(KM.Process):
-  
-    def __init__(self,model_part,params):
+import python_process
+
+# All the processes python processes should be derived from "python_process"
+
+class SPRISMProcess(python_process.PythonProcess):
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/sprism_process.py:20]: I think ususally the python processes derive from the c-process

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_shell_thickness_compute_process.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            ModelPart this_model_part("Main");
+            this_model_part.SetBufferSize(2);
+
+            SolidShellProcessCreateModelPart(this_model_part);
+
+            SolidShellThickComputeProcess thickness_process = SolidShellThickComputeProcess(this_model_part);
+            thickness_process.Execute();
+
+            for (auto& node : this_model_part.Nodes())
+                KRATOS_CHECK_NEAR(node.GetValue(THICKNESS), 1.0, std::numeric_limits<double>::epsilon());
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_shell_thickness_compute_process.cpp:70]: You could use KRATOS_CHECK_EQUAL_DOUBLE here

**Review by philbucher — APPROVED** [review]
Looks ok to me, minor comments

File: applications/StructuralMechanicsApplication/custom_processes/solid_shell_thickness_compute_process.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/solid_shell_thickness_compute_process.cpp:52]: Not right now, but may be

File: applications/StructuralMechanicsApplication/python_scripts/sprism_process.py (hunk 2 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/sprism_process.py:20]: Sure?, I was having serious doubts about that

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_shell_thickness_compute_process.cpp (hunk 3 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_shell_thickness_compute_process.cpp:70]: Ok

File: applications/StructuralMechanicsApplication/python_scripts/sprism_process.py (hunk 2 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/python_scripts/sprism_process.py:20]: I think so

You can check the other python processes ...

**Pull request merged**